pennylane/transforms/metric_tensor.py

```diff
@@ -611,6 +613,10 @@ def processing_fn(results):
         # Get full block-diagonal tensor
         diag_mt = diag_proc_fn(diag_res)
 
+        if qml.active_return():
+            # the off diag tapes only have a single expval measurement
+            off_diag_res = [qml.math.expand_dims(res, 0) for res in off_diag_res]

```

Comment:
Here we expect (1,) shape instead of the new () shape?

Reply:
In the new system the raw results is a list of arrays with shape `()`. In the old system it is a list of arrays with shape `(1,)`. Therefore we expand the dims in the new system to make it consistent with the old system.

Reply:
Thanks!